### PR TITLE
Migrate to command-executor-lib

### PR DESCRIPTION
**Pull request by ep0ll**
### Resolves:
issue: #54

### CODE REVIEW

File: microservices/password-manager/controllers/primistore/primistore-controller.ts [hunk 1]
```diff
@@ -144,7 +147,7 @@ const encryptPasswordHandler = async (req, res, logger) => {
     .readFileSync(charsetPath)
     .toString("utf-8")
     .split("\n")
-    .slice(0, -1);
+    .slice(0, -1).join(" ");
   encryptedPassword = encryptWithCharset(charset, encryptedPassword.value);
```
Comment (ep0ll) [inline microservices/password-manager/controllers/primistore/primistore-controller.ts]: @frankhart2018 can you please have a look at this code, this seems to have incorrect type, what is the expected type goes here?

Comment (ep0ll) [reply microservices/password-manager/controllers/primistore/primistore-controller.ts]: also at line 132

**Comment by anglerfishlyy** [comment]
can you assign me this under GSSOC'24?

File: microservices/password-manager/controllers/primistore/primistore-controller.ts (hunk 1 above)
Comment (frankhart2018) [reply microservices/password-manager/controllers/primistore/primistore-controller.ts]: 132 is a string, 146 is an array of strings

**Review by frankhart2018 — COMMENTED** [review]
Also, I'm confused how node_modules got here, did you add them explicitly? I added those in gitignore.

**Comment by frankhart2018** [comment]
@anglerfishlyy this is already taken by @WYGIN. Feel free to checkout other issues.

**Comment by ep0ll** [comment]
> Also, I'm confused how node_modules got here, did you add them explicitly? I added those in gitignore.

I will take care of it, i believe this is due to not using workspaces!

**Comment by frankhart2018** [comment]
> > Also, I'm confused how node_modules got here, did you add them explicitly? I added those in gitignore.
> 
> I will take care of it, i believe this is due to not using workspaces!

Sounds good.

**Comment by ep0ll** [comment]
@frankhart2018 ready for review

File: package.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (frankhart2018) [inline package.json:1]: Why is this in the root of the project?

File: package-lock.json
Comment (frankhart2018) [inline package-lock.json:1]: Same as package.json this is not required in root.

File: .gitignore [hunk 3]
```diff
@@ -1,2 +1,3 @@
 .DS_Store
-.vscode/
\ No newline at end of file
+.vscode/
+*/node_modules/*
```
Comment (frankhart2018) [inline .gitignore:3]: Not required in root gitignore.

File: microservices/password-manager/controllers/primistore/primistore-controller.ts [hunk 4]
```diff
@@ -35,16 +37,16 @@ const storage = multer.diskStorage({
...
 
-const passwordCreationHandler = async (req, res, logger) => {
+const passwordCreationHandler = async (req: Request, res: Response, logger: winston.Logger) => {
   const password_uid = req.body.identifier;
 
   const { key, iv } = generateAESKeyIV();
   if (
-    key.type === CommandOutputType.Error ||
-    iv.type === CommandOutputType.Error
+    key.type === "error" ||
```
Comment (frankhart2018) [inline microservices/password-manager/controllers/primistore/primistore-controller.ts:45]: The type is an enum, and will return 0 or 1 and not strings.

Comment (frankhart2018) [reply microservices/password-manager/controllers/primistore/primistore-controller.ts:45]: Same for all other command output type comparisons.

File: microservices/password-manager/utils/charset-utils.ts [hunk 5]
```diff
@@ -1,3 +1,5 @@
+import { CommandOutput } from "command-executor-lib/dist/command-output.js";
```
Comment (frankhart2018) [inline microservices/password-manager/utils/charset-utils.ts:1]: Importing an enum that is not exported in the public interface is not a good idea.

File: microservices/password-manager/utils/charset-utils.ts [hunk 6]
```diff
@@ -32,13 +34,14 @@ const mapNumberToPaddedBinaryString = (num: number) => {
     return num.toString(2).padStart(8, "0");
 }
 
-const encryptWithCharset = (charset: string, password: string): string => {
+const encryptWithCharset = (charset: string[], password: string): CommandOutput => {
```
Comment (frankhart2018) [inline microservices/password-manager/utils/charset-utils.ts:37]: Why does this need to return a command output anyway? There is no command execution, everything is done in JS.

File: microservices/password-manager/controllers/primistore/primistore-controller.ts (hunk 4 above)
Comment (frankhart2018) [reply microservices/password-manager/controllers/primistore/primistore-controller.ts:45]: CommandOutputType is already exposed by the latest version of command-executor-lib, do not compare with a string.

File: microservices/password-manager/utils/command-utils.ts [hunk 7]
```diff
@@ -1,27 +1,32 @@
...
-const CommandOutputType = {
-  Success: 0,
-  Error: 1,
-};
+// const CommandOutputType = {
+//   Success: 0,
+//   Error: 1,
+// };
 
 class CommandOutput {
```
Comment (frankhart2018) [inline microservices/password-manager/utils/command-utils.ts:17]: This is not needed anymore.

File: microservices/password-manager/utils/command-utils.ts [hunk 8]
```diff
@@ -1,27 +1,32 @@
...
-  constructor(type, value) {
+  type: CommandOutputType
+  value: string
+  constructor(type: CommandOutputType, value: string) {
     this.type = type;
     this.value = value;
   }
 }
 
 class PipeCommand {
```
Comment (frankhart2018) [inline microservices/password-manager/utils/command-utils.ts:26]: The whole point of using command-executor-lib is I moved all of this logic there, this class is no longer required.

File: microservices/password-manager/utils/command-utils.ts [hunk 9]
```diff
@@ -128,13 +133,13 @@ const runCommandInPipe = (
...
-  return new CommandOutput(CommandOutputType.Success, outputData);
+  return new CommandOutput("success", outputData);
 };
 
-const runScriptInPipe = (scriptFileName, password = null, args = []) => {
+const runScriptInPipe = (scriptFileName: string, password: string = null, args?: string[]) => {
   const scriptPath = path.join("scripts", scriptFileName);
-  const copyScriptResult = runCommand(`cp ${scriptPath} ${PIPE_COMM_DIR}`);
-  if (copyScriptResult.type === CommandOutputType.Error) {
+  const copyScriptResult = new CommandExecutor(`cp ${scriptPath} ${PIPE_COMM_DIR}`, new PipeExecuteStrategy()).execute();
```
Comment (frankhart2018) [inline microservices/password-manager/utils/command-utils.ts:141]: PipeExecutorStrategy is built only using a builder, it should not be built directly.

File: microservices/password-manager/utils/command-utils.ts [hunk 10]
```diff
@@ -143,28 +148,27 @@ const runScriptInPipe = (scriptFileName, password = null, args = []) => {
...
-  runCommand(`rm -f ${newScriptPathOnDevice}`);
+  new CommandExecutor(`rm -f ${newScriptPathOnDevice}`, new PipeExecuteStrategy()).execute();
 
   return runScriptResult;
 };
 
 const generateAESKeyIV = () => {
-  const key = runCommand("openssl rand -hex 32");
-  const iv = runCommand("openssl rand -hex 16");
+  const key = new CommandExecutor("openssl rand -hex 32", new PipeExecuteStrategy()).execute();
```
Comment (frankhart2018) [inline microservices/password-manager/utils/command-utils.ts:159]: Same as above for PipeExecuteStrategy.

File: microservices/password-manager/utils/types.d.ts [hunk 11]
```diff
@@ -0,0 +1,3 @@
+type StringMap = {
```
Comment (frankhart2018) [inline microservices/password-manager/utils/types.d.ts:1]: A types files is not needed for a single type.

**Review by frankhart2018 — CHANGES_REQUESTED** [review]
Please take a look at the changes. Feel free to ping me in discord (`Siddhartha|PA|PrimiStore`) and we can discuss more there.

File: microservices/password-manager/controllers/primistore/primistore-controller.ts (hunk 4 above)
Comment (ep0ll) [reply microservices/password-manager/controllers/primistore/primistore-controller.ts:45]: I changed the type into string, and typescript autocompletion suggests `success`, `error`, and I believe there is no reason to use enums!

File: package.json (hunk 2 above)
Comment (ep0ll) [reply package.json:1]: Should I migrate the project to npm workspace then?

Comment (frankhart2018) [reply package.json:1]: Let's discuss all of the changes in discord :)

**Comment by frankhart2018** [comment]
Closing this. Please re-open if you want to wrap this up.

**Comment by ep0ll** [comment]
> Closing this. Please re-open if you want to wrap this up.

For sure :)

**Pull request closed without merging**